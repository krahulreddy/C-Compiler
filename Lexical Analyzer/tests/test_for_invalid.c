#include <stdio.h>

void main()
{
    int i;

    for(i = 10; i <=20; i++)
    {
        printf("Hello");
        printf("%s ", "i);
        
        for(i = 0; i <= 10; i++);
    }

}