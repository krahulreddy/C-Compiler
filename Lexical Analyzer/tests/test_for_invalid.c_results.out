Line			Lexeme			Token Class
1		#include <stdio.h> 		Pre Processor directive

2
3			void 			KEYWORD
			main 			KEYWORD
			( 			OPENING BRACKETS
			) 			CLOSING BRACKETS

4			{ 			OPENING BRACES

5			int 			KEYWORD
			i 			IDENTIFIER
			; 			SEMICOLON DELIMITER

6
7			for 			KEYWORD
			( 			OPENING BRACKETS
			i 			IDENTIFIER
			= 			OPERATOR
			10 			NUMBER CONSTANT
			; 			SEMICOLON DELIMITER
			i 			IDENTIFIER
			<= 			OPERATOR
			20 			NUMBER CONSTANT
			; 			SEMICOLON DELIMITER
			i 			IDENTIFIER
			++ 			OPERATOR
			) 			CLOSING BRACKETS

8			{ 			OPENING BRACES

9			printf 			IDENTIFIER
			( 			OPENING BRACKETS
			"Hello" 			STRING CONSTANT
			) 			CLOSING BRACKETS
			; 			SEMICOLON DELIMITER

10			printf 			IDENTIFIER
			( 			OPENING BRACKETS
			"%s " 			STRING CONSTANT
			, 			COMMA DELIMITER
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
ERROR: INCOMPLETE_STRING at line no. 10
	"
	^^
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX


SYMBOL TABLE

	i		IDENTIFIER
	for		KEYWORD
	int		KEYWORD
	main		KEYWORD
	printf		IDENTIFIER
	void		KEYWORD


CONSTANT TABLE

	"Hello"		STRING CONSTANT
	10		NUMBER CONSTANT
	20		NUMBER CONSTANT
	"%s "		STRING CONSTANT
